lib/external-parsers/base.ts

```diff
@@ -0,0 +1,98 @@
+import fs from 'fs';
+import path from 'path';
+
+import { TypicalExecutionFunc, UnprocessedExecResult } from '../../types/execution/execution.interfaces';
+import { IParseFilters } from '../../types/features/filters.interfaces';
+import { maskRootdir } from '../utils';
+
+import { IExternalParser, IParsedAsmResult } from './external-parser.interface';
+
+const starterScriptName = 'dump-and-parse.sh';
+
+export class ExternalParserBase implements IExternalParser {
+    private objdumperPath: string;
```

Comment:
My IDE is giving me hints that `objdumpedPath`, `parserPath` and `execFunc` can be made `readonly`

Reply:
Makes sense